R/utils.R

```diff
@@ -116,3 +133,88 @@ recycle <- function(x, n, name = deparse(substitute(x))) {
     stop(sprintf("Invalid length for '%s', expected 1 or %d", name, n))
   }
 }
+
+
+is_3d_array <- function(x) {
+  !is.na(nlayer(x))
+}
+
+
+## Copied from ncol
+nlayer <- function(x) {
+  dim(x)[[3L]]
+}
+
+
+## Copied from NCOL
+NLAYER <- function(x) { # nolint
+  d <- dim(x)
+  if (length(d) > 2L) {
+    d[3L]
+  } else {
+    1L
+  }
+}
+
+
+## Copied from colnames
+layernames <- function(x, do.NULL = TRUE, prefix = "layer") { # nolint
+    dn <- dimnames(x)
+    if (!is.null(dn[[3L]]))
+        dn[[3L]]
+    else {
+      if (do.NULL) {
+        NULL
+      } else {
+        paste0(prefix, seq_len(NLAYER(x)))
+      }
+    }
+}
+
+
+## Copied from colnames<-
+`layernames<-` <- function(x, value) { # nolint
+  dn <- dimnames(x)
+  if (is.null(dn)) {
+    if (is.null(value)) {
+      return(x)
+    }
+    nd <- length(dim(x))
+    if (nd < 3L) {
+      stop("attempt to set 'layernames' on an object with less than three
+      dimensions")
+    }
+      dn <- vector("list", nd)
+  }
+  if (length(dn) < 3L) {
+    stop("attempt to set 'colnames' on an object with less than three
+    dimensions")
+  }
+  if (is.null(value)) {
+    dn[3L] <- list(NULL)
+  } else {
+    dn[[3L]] <- value
+  }
+  dimnames(x) <- dn
+  x
+}
+
+
+set_layernames <- function(m, nms) {
+  layernames(m) <- nms
+  m
+}
+
+lbind <- function(arrays) {
```

Comment:
this is array_bind I think, though that is likely more efficient as it avoids the do.call; binding on the last array dimension is the easiest